akka-cluster-sharding/src/main/scala/akka/cluster/sharding/protobuf/ClusterShardingMessageSerializer.scala

```diff
@@ -373,9 +376,10 @@ private[akka] class ClusterShardingMessageSerializer(val system: ExtendedActorSy
     shardRegionStatsFromProto(parsed)
   }
 
-  private def shardRegionStatsFromProto(parsed: ClusterShardingMessages.ShardRegionStats) = {
+  private def shardRegionStatsFromProto(parsed: ClusterShardingMessages.ShardRegionStats): ShardRegionStats = {
     val stats: Map[String, Int] = parsed.getStatsList.asScala.iterator.map(e => e.getKey -> e.getValue).toMap
-    ShardRegionStats(stats)
+    val failed: Set[String] = parsed.getFailedList.asScala.toSet
```

Comment:
Looks like this is correctly initialized to the empty list when no `failed` field has been found, so sending these messages from old nodes to new nodes should work :+1: